src/v2/models/AppState.js

```diff
@@ -29,15 +29,19 @@ export default Model.extend({
 
   derived: {
     factorProfile: {
-      deps: ['factor'],
-      fn (factor = {}) {
-        return factor.profile || {};
+      // While we're moving toward `authenticator` platform, but still
+      // need to support `factor` for certain period.
+      // Could remove `factor` after it's deprecated completely.
+      deps: ['factor', 'authenticator'],
```

Comment:
seems like views would still be referring to factorProfile which is a good idea for now but eventually I would call it authenticator instance.

Reply:
I'm not sure if I understand your statement.
`factor` and `authenticator` won't exists at same time. how `view refer to factorProfile` if it doesn't exists. When API responds `authenticator`, you'll get `authenticator.profile`.